pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -209,13 +211,44 @@ func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *v1.Pod) erro
 	pod.Spec.Volumes = append(pod.Spec.Volumes, podVolumes...)
 
 	// Inject credentials
-	if err := mi.credentialBuilder.CreateSecretVolumeAndEnv(
-		pod.Namespace,
-		pod.Spec.ServiceAccountName,
-		initContainer,
-		&pod.Spec.Volumes,
-	); err != nil {
-		return err
+	hasStorageSpec, ok := pod.ObjectMeta.Annotations[constants.StorageSpecAnnotationKey]
+	storageKey, storageKey_exist := pod.ObjectMeta.Annotations[constants.StorageSpecKeyAnnotationKey]
```

Comment:
nit: Better to use standard go camelcase convention e.g. `storageKeyExists`

Reply:
done